src/main/java/com/oltpbenchmark/benchmarks/auctionmark/AuctionMarkWorker.java

```diff
@@ -447,8 +447,24 @@ protected boolean executeCloseAuctions(Connection conn, CloseAuctions proc) thro
         Timestamp startTime = profile.getLastCloseAuctionsTime();
         Timestamp endTime = profile.updateAndGetLastCloseAuctionsTime();
 
-        List<Object[]> results = proc.run(conn, benchmarkTimes, startTime, endTime);
-
+        List<Object[]> results = null;
+        boolean done = false;
+        // Retry here the close auction procedure if it fails due to a transactional conflict, since
+        // the original submitted procedured has a different type.
+        while (!done) {
+            try {
+                results = proc.run(conn, benchmarkTimes, startTime, endTime);
+                done = true;
+            }
+            catch (SQLException e)  {
+                if (e.getSQLState().startsWith("40")) {
+                    conn.rollback();
+                }
```

Comment:
Can this get stuck in an infinite loop?

Reply:
I don't think so, because sooner or later the transaction will be able to complete. Any other type of error other than transactional conflicts will raise an exception to the worker.